src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -11,35 +11,50 @@ import SCREENS from '../../../SCREENS';
 import ReportScreen from '../../../pages/home/ReportScreen';
 import SidebarScreen from '../../../pages/home/sidebar/SidebarScreen';
 import BaseDrawerNavigator from './BaseDrawerNavigator';
-import {findLastAccessedReport} from '../../reportUtils';
+import {findLastAccessedReport, findConciergeDMReport} from '../../reportUtils';
+import _ from "underscore";
+import * as API from "../../API";
 
 const propTypes = {
     /** Available reports that would be displayed in this navigator */
     reports: PropTypes.objectOf(PropTypes.shape({
         reportID: PropTypes.number,
     })),
+
+    triggerWelcomeLocation: PropTypes.string,
 };
 
 const defaultProps = {
     reports: {},
+    triggerWelcomeLocation: '',
 };
 
 
-const getInitialReportScreenParams = (reports) => {
-    const last = findLastAccessedReport(reports);
+const getInitialReportScreenParams = (reports, triggerWelcomeLocation) => {
+    let report;
+
+    if (triggerWelcomeLocation && !_.isEmpty(reports)) {
+        report = findConciergeDMReport(reports);
+        // API.TriggerWelcome({location: triggerWelcomeLocation});
+        // Onyx.clear(ONYXKEYS.TRIGGER_WELCOME_LOCATION);
```

Comment:
You might want to check out the source of `Onyx.clear()` [here](https://github.com/Expensify/react-native-onyx/blob/master/lib/Onyx.js#L637) because you're definitely holding it wrong 😁 

There are no arguments for `clear()` and it's meant to empty out the entire store (used on sign-out). 

I think what you are intending to do is `Onyx.set(ONYXKEYS.TRIGGER_WELCOME_LOCATION, null)`

This is some more code that should be in an action (but maybe you are just testing it here first).

I am a little bit confused on why there is code being added to `MainDrawerNavigator.js`, so maybe I am missing something. I don't think there should be any code in here and it seems like this code should all be in `TriggerWelcomePage.js`.

Reply:
oh lol i was definitely holding clear wrong. So dumb 🤦  it worked so easily with `Onyx.set(ONYXKEYS.TRIGGER_WELCOME_LOCATION, null)`

> I am a little bit confused on why there is code being added to MainDrawerNavigator.js, so maybe I am missing something. I don't think there should be any code in here and it seems like this code should all be in TriggerWelcomePage.js.

Yeah if I could have everything in TriggerWelcomePage.js that would be great but I'm not sure how can I. So the tricky flow is when the user is not logged in. So say a user scans a QR code which redirects them to http://expensify.cash/trigger/welcome/sfo. Thats going to hit the TriggerWelcomePage and I can save the data that I want and redirect the user to signin screen when there is no authentication. But once the user signs in the route won't be `trigger/welcome/sfo` so I can't run any of TriggerWelcomePage's logic and hence I have it in MainDrawerNavigator.js. If you think there is something easier I'm missing so that I can have the logic in TriggerWelcomePage let me know.

Reply:
Ah! Yes, I'm following you now. If you look at the solution that I did for the e.com <--> e.cash link, you'll see how I did that for the "new workspace" deep link: https://github.com/Expensify/Expensify.cash/blob/04e830b256b0141027c067c7e89c091306a862d2/src/Expensify.js#L124-L126

I would do something similar in your case, and just navigate back to `TriggerWelcomePage` once they are signed in. Then, the `AuthScreens` version of that page should run all the logic that you already have.

Something else to consider, and I was careful not to preoptimize anything so it could be refactored... but since this is the second time we are doing a deep-link like this, it's probably time to consider making this a more generic solution. What would you think about changing `SESSION.redirectToWorkspaceNewAfterSignIn` from a boolean, into a string that holds the route to redirect to. It would look something like this:

```
            if (lodashGet(this.props, 'session.redirectToRouteAfterSignIn','')) {
                Navigation.navigate(this.props.session.redirectToRouteAfterSignIn;
            }
```

Reply:
oh wow, thats so cool and yeah it does look like exactly what can help me here. Yeah I'll check out `SESSION.redirectToWorkspaceNewAfterSignIn` with a string param. Thanks tim.